examples/data_loaders/data_loaders.ipynb

```diff
@@ -0,0 +1,1494 @@
+    "    pass\n",
+    "\n",
+    "\n",
+    "@does(_sum_series)\n",
+    "def total_churn(mens_churn: pd.Series, womens_churn: pd.Series) -> pd.Series:\n",
+    "    pass\n",
+    "\n",
+    "\n",
+    "def total_customers(total_signups: pd.Series, total_churn: pd.Series) -> pd.Series:\n",
+    "    customer_deltas = total_signups + total_churn\n",
+    "    return customer_deltas.cumsum()\n",
+    "\n",
+    "\n",
+    "def acquisition_cost(total_marketing_spend: pd.Series, total_signups: pd.Series) -> pd.Series:\n",
+    "    return total_marketing_spend / total_signups\n",
+    "\n",
+    "\n",
+    "@parameterize(\n",
+    "    twitter_spend_smoothed={\"lookback_days\": value(7), \"spend\": source(\"twitter_spend\")},\n",
+    "    facebook_spend_smoothed={\"lookback_days\": value(7), \"spend\": source(\"facebook_spend\")},\n",
+    "    radio_spend_smoothed={\"lookback_days\": value(21), \"spend\": source(\"radio_spend\")},\n",
+    "    tv_spend_smoothed={\"lookback_days\": value(21), \"spend\": source(\"tv_spend\")},\n",
+    "    billboards_spend_smoothed={\"lookback_days\": value(7), \"spend\": source(\"billboards_spend\")},\n",
+    "    youtube_spend_smoothed={\"lookback_days\": value(7), \"spend\": source(\"twitter_spend\")},\n",
+    ")\n",
+    "def spend_smoothed(lookback_days: int, spend: pd.Series) -> pd.Series:\n",
+    "    \"\"\"{spend} smoothed by {lookback_days}. Might want to smooth different ad spends differently,\n",
+    "    figuring that it takes different amounts of time to get to the customer. A cheap hack at determining\n",
+    "    auto-correlation of a series -- this should be a parameter in a model,\n",
+    "    but this is to demonstrate the framework\n",
+    "\n",
+    "    :param lookback_days: Days to smooth over\n",
+    "    :param spend: Spend source\n",
+    "    :return:\n",
+    "    \"\"\"\n",
+    "    return spend.rolling(window=lookback_days).mean().fillna(0)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "VARS = [\n",
+    "    \"total_signups\",\n",
+    "    \"total_churn\",\n",
+    "    \"total_marketing_spend\",\n",
+    "    \"acquisition_cost\",\n",
+    "    \"twitter_spend_smoothed\",\n",
+    "    \"facebook_spend_smoothed\",\n",
+    "    \"radio_spend_smoothed\",\n",
+    "    \"tv_spend_smoothed\",\n",
+    "    \"billboards_spend_smoothed\",\n",
+    "    \"youtube_spend_smoothed\",\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Execution for duckdb data\n",
+    "\n",
+    "driver = hamilton.driver.Driver(\n",
+    "    {\"db_path\": \"./test_data/database.duckdb\"}, load_data_duckdb, prep_data\n",
+    ")\n",
+    "print(driver.execute(VARS))\n",
+    "duckdb_execution_graph = driver.visualize_execution(VARS)\n",
+    "display(duckdb_execution_graph)"
```

Comment:
Can you use the newer builder pattern please?
e.g.
```python
from hamilton import base
import hamilton.driver

driver = (hamilton.driver.Builder()
      .with_modules(load_data_duckdb, prep_data)
      .with_adapters(base.PandasDataFrameResult()
      .build()
)
```

then for the visualization pass in inputs with the dictionary:

```python
duckdb_execution_graph = driver.visualize_execution(VARS, inputs={"db_path": "./test_data/database.duckdb"})
```

Reply:
Thank you @skrawcz, I have implemented the changes suggested.